### PR TITLE
Feature/wagtail 3 support

### PR DESCRIPTION
**Pull request by fabienheureux**
This PR brings wagtail 3 support to Grapple.

Questions / comments :
- [ ] I had failing tests following the upgrade, because of the search query `"searchQuery": "t"` in the tests, that returned no pages. Could it be because of changes in Wagtail ? I updated the query to search for `"searchQuery": "Test"` and it works properly. 
- [ ] I did not update wagtail.core paths in migrations, should we update these as well ? 
- [ ] Flake8 complained about C417 error, I fixed these in this PR

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       max-parallel: 4
       matrix:
         python-version: ['3.7', '3.8', '3.9', '3.10']
-        wagtail-version: ['2.14', '2.15', '2.16']
+        wagtail-version: ['2.14', '2.15', '2.16', '3.0']
```
Comment (zerolab) [inline .github/workflows/ci.yml:32]: let's drop support for Wagtail < 2.15

File: example/home/models.py [hunk 2]
```diff
@@ -36,6 +42,11 @@
 )
 from grapple.utils import resolve_paginated_queryset
 
+try:
```
Comment (zerolab) [inline example/home/models.py:45]: nitpic: I'd prefer a more explicit conditional usage in the panel definitions (i.e. `FieldPanel() if WAGTAIL_VERSION >= (3, 0) else StreamFieldPanel`

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 Django>=3.0,<4.0
-wagtail>=2.14,<2.17
+wagtail>=2.14,<4.0
```
Comment (zerolab) [inline requirements.txt:2]:
```suggestion
wagtail>=2.15,<4.0
```

File: setup.cfg [hunk 4]
```diff
@@ -40,7 +41,7 @@ include_package_data = true
 packages = find:
 install_requires =
     Django>=3.0,<4.0
-    wagtail>=2.14, <2.17
+    wagtail>=2.14, <4.0
```
Comment (zerolab) [inline setup.cfg:44]:
```suggestion
    wagtail>=2.15, <4.0
```

**Review by zerolab — CHANGES_REQUESTED** [review]
Lot of good stuff, @fabienheureux 
Thank you for this. And sorry it took so long to follow up.

A few changes and we should be good to go

**Comment by fabienheureux** [comment]
Hey @zerolab , thanks for the review !
I have updated the PR according to your feedback, let me know if it requires more changes.

**Review by zerolab — APPROVED** [review]
LGTM. Thank you @fabienheureux

**Pull request merged**